### PR TITLE
[french_zhjay] New keyboard for (ZHJAY) French typewriter

### PR DESCRIPTION
**Pull request by mareuo**

### CODE REVIEW

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Comment by mareuo** [comment]
zhjay

File: release/f/french_zhjay/source/french_zhjay.kps [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        <Language ID="fr">French</Language>
+      </Languages>
+      <Examples>
+        <Example ID="fr" Keys="F r q n 9 q i s" Text="Français" Note="Name of Language"/>
+      </Examples>
+    </Keyboard>
+  </Keyboards>
+  <RelatedPackages>
+    <RelatedPackage ID="french" Relationship="deprecates"/>
+    <RelatedPackage ID="kbdfr" Relationship="deprecates"/>
```
Comment (darcywong00) [inline release/f/french_zhjay/source/french_zhjay.kps:97]: Is this related to basic_kbdfr?  (which already deprecated "french" and "kbdfr")

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Comment by DavidLRowe** [comment]
Thanks for this contribution.
For the desktop layout, there are some issues with deadkeys.
- Documentation lists circumflex, dieresis, grave and tilde as deadkeys.
- None of these are shown on the on-screen keyboard (OSK)
- Shift+² produces ~ (even though the OSK has a blank key), but the tilde doesn't operate as deadkey.
- The key between "," and "$" is blank, but functions correctly as the circumflex deadkey.
- That key on the shift layer is blank, but functions correctly as the dieresis deadkey.

On the touch layout:
- The shift (up-arrow) key doesn't work to switch to shift layer (neither on the default nor the rightalt layer)
- The circumflex deadkey on the default layer (and presumably the dieresis deadkey on the shift layer) are not marked
- The use of dead keys on a touch layout is a little odd. You might consider adding long press keys to the base keys (that is, on the "a" key, add long press keys for a+tilde, a+grave, a+dieresis, a+circumflex, and so on).

**Comment by mareuo** [comment]
I worked with basic french layout, didnt change deadkeys

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Comment by DavidLRowe** [comment]
I realize that you may still be making changes to this keyboard, but I wanted to point out that adding a key with an identifier like "T_new_1015" for a longpress "â" won't produce any output unless there's a corresponding rule in the .kmn file for [T_new_1015] that produces the desired output. (Note that what comes after the "T_" can be any valid name, so "T_a_circumflex" might be more easily understood in this case.)

Alternatively, you can use the "U_xxxx" convention, where Keyman will generate the corresponding U+xxxx character (provided nothing in the .kmn file changes [U_xxxx] to something else!). So, for the a-circumflex you could set the "id" field to "U_00E2" and wouldn't need to add anything to the .kmn file.

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Comment by DavidLRowe** [comment]
Thanks for your updates. There are still a few things to correct and some things you may want to consider changing.

On the touch layout, shift layer
- The shift key doesn't take you back to the default layer. You'll need to change the "nextlayer" field to switch to the default layer.
- The leftmost key on the third row is blank, but produces "~". You should either display "~" on the key, or make the key produce no output.
- One of the longpress keys on the "I" key displays and produces "ï", whereas "Ï" would be expected.

It's not clear from the documentation for the desktop layout how one would type the grave and tilde deadkeys, since they are blank keys on the keyboard image for the RightAlt layer. At the least, the image should have a tilde and a grave for those keys. (Some people prefer to use a dotted circle (U+25CC ◌ ) plus a combining accent (U+0300 for combining grave or U+0303 for combining tilded) to indicate that the key is used to create diacritics.)

Other changes to consider:
On the touch layout, default layer
- There's an empty longpress key on the "a" key. It does no harm, but it would probably be better to remove it.
On the touch layout, rightalt layer
- The deadkeys for the tilde and grave are not labeled. Since you've added these combinations as longpress keys, you could leave this as is, but it runs counter to the idea that a touch layout is showing you what each key will do.

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Comment by mareuo** [comment]
This keyboard is ready

**Comment by DavidLRowe** [comment]
Touch layout, default layer:
ÿ (longpress key on y): change ID from U+00FF to U_00FF
œ (longpress key on o): change ID from U+0153 to U_0153

Touch layour, shift layer:
Ÿ (longpress key on Y): change ID from U+0178 to U_0178
Œ (longpress key on O): change ID from U+0152 to U_0152
Shift key: needs to specify "default" in the Next Layer field.

**Review by DavidLRowe — APPROVED** [review]
